src/libraries/System.Private.CoreLib/src/System/IO/StringReader.cs

```diff
@@ -160,10 +165,10 @@ public override string ReadToEnd()
             }
             else
             {
-                s = _s.Substring(_pos, _length - _pos);
+                s = _s.Substring(_pos);
             }
 
-            _pos = _length;
+            _pos = _s.Length;
             return s;
```

Comment:
Ditto.  It seems like this method could be:
```C#
string? s = _s;
if (s == null)
{
    ThrowObjectDisposedException_ReaderClosed();
}

int pos = _pos;
_pos = s.Length;

if (pos != 0)
{
    s = s.Substring(pos);
}

return s;
```

Reply:
copied as suggested